src/_tests/integration/integration-utils.js

```diff
@@ -1,5 +1,40 @@
+const waitForFn = async ({ fn, interval = 2000, timeout = 10000 }) => {
+  const readyState = new Promise(resolve => {
+    const start = Date.now()
+    const intervalId = setInterval(() => {
+      const success = fn()
+      success && resolve({ success, intervalId })
+      Date.now() - start > timeout && resolve({ intervalId })
```

Comment:
If this times out, we should abort the test anyway, so perhaps we should reject instead. (If we do that we'll have to modify the handling of `clearInterval`)